book_maker/loader/epub_loader.py

```diff
@@ -73,6 +91,28 @@ def _make_new_book(self, book):
         return new_book
 
     def make_bilingual_book(self):
+        def deal_new(p, wait_p_list):
+            ret = deal_old(wait_p_list)
+            new_p = copy(p)
+            new_p.string = self.translate_model.translate(p.text)
+            p.insert_after(new_p)
+            return ret
+
+        def deal_old(wait_p_list):
+            if len(wait_p_list) == 0:
+                return []
+
+            result_txt_list = self.translate_model.translate_list(wait_p_list)
+
+            for i in range(0, len(wait_p_list)):
+                if i < len(result_txt_list):
+                    p = wait_p_list[i]
+                    new_p = copy(p)
+                    new_p.string = result_txt_list[i]
+                    p.insert_after(new_p)
+
+            return []
```

Comment:
I still don't understand why we need to return [], as it always seems to be an empty list. Could you explain it again?

Reply:
I added this because the editor warned: `"append" is not a known member of "None" (Pyright reportOptionalMemberAccess)`

```
wait_p_list = deal_old(wait_p_list)
wait_p_list.append(p)
```

return None, append may get error

Reply:
```
deal_old(wait_p_list)
wait_p_list.append(p)
``` 
So why not use this code.Do not assign `deal_old` result to `wait_p_list`?

Reply:
According to the logic, the `wait_p_list` should be emptied. Do you mean that the return value is not used and it is emptied directly in the function?

Reply:
Yes, I think this code logic is clearer，but that can depend on your preferences :D

Reply:
In fact, I first gave `wait_p_list = []` directly in the function, but the `wait_p_list` didn't change. I don't know why. I thought it was a reference type and should be changed, but it didn't... So I used the return value helplessly

Reply:
Maybe `wait_p_list.clear()` will work.

Reply:
Thanks, I learned this useful method,